goal_src/pc/pckernel.gc

```diff
@@ -760,15 +805,28 @@
 
 (defmethod new pc-settings ((allocation symbol) (type-to-make type))
   "make a new pc-settings"
-
   (let ((obj (object-new allocation type-to-make (the-as int (-> type-to-make size)))))
-
     (reset obj)
-
-    obj
-    )
-  )
-
+    ;; auto load settings if available
+    ;; if saved settings are corrupted, reset them
+    (clear *pc-temp-string-1*)
+    (format *pc-temp-string-1* "~S/pc-settings.gc" *pc-settings-folder*)
+
+    (if (pc-filepath-exists? *pc-temp-string-1*)
+      (begin
+        (format 0 "[PC] PC Settings found at '~S'...loading!~%" *pc-temp-string-1*)
+        (if (read-from-file obj *pc-temp-string-1*)
+          (return obj)
+          (begin
+            (format 0 "[PC] PC Settings found at '~S' but could not be loaded, re-creating with defaults!~%" *pc-temp-string-1*)
+            (reset obj))))
```

Comment:
hmm yes that function should probably be altered to not overwrite the settings in case an error occurred, I'll have to fix that later.

Reply:
I think resetting the settings back to defaults when they are not valid is sensible -- we aren't wiping save-file data or something important.  It's better imo than crashing.